sdk/core/azure-core/src/test/java/com/azure/core/util/serializer/JacksonAdapterIT.java

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.util.serializer;
+
+import com.azure.core.http.HttpHeaders;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+/**
+ * Integration tests for {@link JacksonAdapter}.
+ */
+public class JacksonAdapterIT {
+    @Test
+    public void stronglyTypedHeadersClassIsLazilyDeserialized() throws IOException {
+        final String expectedValue = "the-string-value";
+
+        HttpHeaders rawHeaders = new HttpHeaders().set("the-string", "the-string-value");
+
+        StronglyTypedHeaders stronglyTypedHeaders = JacksonAdapter.createDefaultSerializerAdapter()
+            .deserialize(rawHeaders, StronglyTypedHeaders.class);
+
+        assertNull(stronglyTypedHeaders.theString);
+        assertEquals(expectedValue, stronglyTypedHeaders.getTheString());
+    }
+
+    public static final class StronglyTypedHeaders {
+        private final HttpHeaders rawHeaders;
+
+        private boolean hasTheStringBeenDeserialized;
+        private String theString;
+
+        public StronglyTypedHeaders(HttpHeaders rawHeaders) {
+            this.rawHeaders = rawHeaders;
+        }
+
+        String getTheString() {
+            if (hasTheStringBeenDeserialized) {
+                return theString;
+            }
+
+            theString = rawHeaders.getValue("the-string");
+            hasTheStringBeenDeserialized = true;
+
+            return theString;
+        }
```

Comment:
I remember similar idea being discussed for .NET. The concern there was that:

```
try {
  TypedHeaders foo = client.getFoo();
} catch (Exception e) {
// kaboom
}
foo.getTheString();
```
would now not work if deserialization exception would be thrown in different place. 
Should we preserve that behavior in APIs where typed headers are emitted to the user?

Reply:
Looks like `eagerDeserializationUsed` is going to solve this per snippet in other comment?

Reply:
`eagerDeserializationUsed` only exists if there was a reflection issue when attempting to use the new pattern and Jackson Databind is used instead. That said, we could add contextual information into the call or on the class itself to eagerly deserialize if it is a type returned to the caller